main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
+    with open("setup.json", encoding="utf-8") as setup_file:
+        settings = json.load(setup_file)
+
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
+    """Lifespan events.
+
+    This function is set as the lifespan of the application.
+
+    The code before 'yield' is executed once before the application starts.
+    The code after 'yield' is executed when the application is shutting down.
+    """
+    load_setup_file()
+    yield
+
+
+app = FastAPI(lifespan=lifespan)
+
+
+@app.get("/ls/")
+async def list_directory(directory: str = ""):
+    """Get the list of elements in the given path.
+
+    The 'master_path' is used for the prefix of the path.
```

Comment:
Not a big deal, but `"master_path"` seems more consistent with our quote convention.

Reply:
In addition, let's clarify that it is the "master_path" in the configuration file.